# Names
1. 169084 - Ali Omar
2. 168945 - Denzel Ngare
3. 169819 - Sumeiya Abass
4. 169977 - Suweyba Ahmed
5. 150496 - Emran Abdi Ahmed
6. 150350 - Alio Ilyas Abdi

# Netflix Titles Data Cleaning

In [1]:
import pandas as pd

# Load your cleaned dataset
df = pd.read_csv("netflix_titles_clean_utf8.csv")

# Display the first few rows
df.head()

show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water              NaN   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans              NaN   
4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA   1 Season   
3  September 24, 2021          2021  TV-MA   1 Season   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

## 1.	How many rows and columns are contained in the dataset? 

In [5]:
df.shape 

(8807, 12)

##### The dataset contains 8807 rows and 12 columns

## 2.	What datatypes are contained in the dataset? 

In [4]:
df.dtypes

show_id         object
type            object
title           object
director        object
cast            object
country         object
date_added      object
release_year     int64
rating          object
duration        object
listed_in       object
description     object
dtype: object

##### Answer:
Most of the columns in the dataset are of type `object`.  
The `release_year` column is of type `int64`.

## 3.	Is the dataset complete? i.e no missing values? 

In [6]:
df.isnull().sum()

show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration           3
listed_in          0
description        0
dtype: int64

##### Answer:
The dataset is not complete — some columns contain missing values.

- `director` has 2,634 missing entries  
- `cast` has 825 missing entries  
- `country` has 831 missing entries  
- `date_added` has 10 missing entries  
- `rating` has 4 missing entries  
- `duration` has 3 missing entries  

All other columns are complete (no missing values).


## 4.	List the number of Movies and TV Shows among the last 20 rows of the dataset.

In [7]:
df.tail(20)['type'].value_counts()

type
Movie      15
TV Show     5
Name: count, dtype: int64

##### Answer:
Among the last 20 rows of the dataset:
- There are 15 Movies
- There are 5 TV Shows

## 5.	Give the number of all Movies originating from South Africa

In [8]:
df[(df['country'] == 'South Africa') & (df['type'] == 'Movie')].shape[0]

24

##### There are 24 movies from South Africa.

## 6.	What is the description of the movie in the 12th row?

In [9]:
df.loc[11, 'description']

'Struggling to earn a living in Bangkok, a man joins an emergency rescue service and realizes he must unravel a citywide conspiracy.'

##### The description is: 'Struggling to earn a living in Bangkok, a man joins an emergency rescue service and realizes he must unravel a citywide conspiracy.'

## 7.	From these South African movies, list only those released in 2021

In [10]:
df[(df['country'] == 'South Africa') & 
   (df['release_year'] == 2021) & 
   (df['type'] == 'Movie')]

show_id   type                         title       director  \
293     s294  Movie                          Slay      Adze Ugah   
764     s765  Movie  Trippin' with the Kandasamys  Jayan Moodley   
881     s882  Movie                I Am All Girls  Donovan Marsh   
1172   s1173  Movie       Loyiso Gola: Unlearning  Kagiso Lediga   

                                                   cast       country  \
293   Enhle Mbali, Ramsey Nouah, Dawn Thandeka Kang,...  South Africa   
764   Jailoshini Naidoo, Maeshni Naicker, Madhushan ...  South Africa   
881   Erica Wessels, Hlubi Mboya, Deon Lotz, Brendon...  South Africa   
1172                                        Loyiso Gola  South Africa   

          date_added  release_year rating duration  \
293   August 8, 2021          2021  TV-MA   86 min   
764     June 4, 2021          2021  TV-14   94 min   
881     May 14, 2021          2021  TV-MA  107 min   
1172  March 23, 2021          2021  TV-MA   60 min   

                                            listed_in  \
293   Comedies, International Movies, Romantic Movies   
764   Comedies, International Movies, Romantic Movies   
881           Dramas, International Movies, Thrillers   
1172                                  Stand-Up Comedy   

                                            description  
293   In pursuit of both success and validation, a g...  
764   To rekindle their marriages, best friends-turn...  
881   A relentless detective finds common ground wit...  
1172  South African comedian Loyiso Gola serves up f...

##### Answer:
There are 4 South African movies released in 2021:
1. Slay - Directed by Adze Ugah
2. Trippin' with the Kandasamys - Directed by Jayan Moodley
3. I Am All Girls - Directed by Donovan Marsh
4. Loyiso Gola: Unlearning - Directed by Kagiso Lediga

## 8.	List the entire cast of all movies released in 2021.

In [11]:
df[(df['type'] == 'Movie') & (df['release_year'] == 2021)]['cast']

6       Vanessa Hudgens, Kimiko Glenn, James Marsden, ...
9       Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...
12      Luna Wedler, Jannis Niewöhner, Milan Peschel, ...
13      Klara Castanho, Lucca Picon, Júlia Gomes, Marc...
18      Freida Pinto, Logan Marshall-Green, Robert Joh...
                              ...                        
1425                                                  NaN
1431    Engin Günaydın, Haluk Bilginer, Binnur Kaya, Ö...
1436    Juan Sebastián Gutiérrez, Pablo Cano, Pablo Me...
1466                                                  NaN
1468             Cha In-pyo, Cho Dal-hwan, Song Jae-ryong
Name: cast, Length: 277, dtype: object

##### Answer:
The dataset contains 277 movies released in 2021, each with its corresponding cast listed above.  
Examples include:  
- Vanessa Hudgens, Kimiko Glenn, James Marsden  
- Melissa McCarthy, Chris O'Dowd, Kevin Kline 
- Luna Wedler, Jannis Niewöhner, Milan Peschel and many more.

Some rows have missing cast information (shown as `NaN`), meaning no cast data was recorded for those titles.


## Team Photo
![Team Photo](team_photo.png)